scan/src/utils/Route.re

```diff
@@ -39,87 +39,165 @@ type t =
   | ValidatorHomePage
   | ValidatorIndexPage(Address.t, validator_tab_t);
 
+let regexes = [
+  ("^$" |> Js.Re.fromString, _ => HomePage),
+  ("^blocks$" |> Js.Re.fromString, _ => BlockHomePage),
+  ("^scripts$" |> Js.Re.fromString, _ => OracleScriptHomePage),
+  ("^sources$" |> Js.Re.fromString, _ => DataSourceHomePage),
+  ("^txs$" |> Js.Re.fromString, _ => TxHomePage),
+  (
+    "^tx$" |> Js.Re.fromString,
+    keys => {
+      switch (keys->Belt_List.get(1)) {
+      | Some(hash) => TxIndexPage(Hash.fromHex(hash))
+      | None => NotFound
+      };
+    },
+  ),
+  ("^validators$" |> Js.Re.fromString, _ => ValidatorHomePage),
+  (
+    "B([0-9]+)$" |> Js.Re.fromString,
+    keys => {
+      switch (keys->Belt_List.get(1)->Belt_Option.getWithDefault("")->int_of_string_opt) {
+      | Some(height) => BlockIndexPage(height)
+      | None => NotFound
+      };
+    },
+  ),
+  (
+    "D([0-9]+)$" |> Js.Re.fromString,
+    keys => {
+      switch (
```

Comment:
IMO. Makes more sense to make this return
```
DataSourceIndexPage(
    keys->Belt_List.getExn(1)->int_of_string, 
    dataSourceTab(keys->Belt_List.getExn(2)),
)
```
and define `dataSourceTab` function separately.

Reply:
What should we do if keys->Belt_List.getExn(2) not in ["","code","requests","revisions"]

Reply:
I currently solve this by using
    keys =>
      dataSourceTab(keys->Belt_List.getExn(2))
      ->Belt_Option.mapWithDefault(NotFound, tab =>
          DataSourceIndexPage(keys->Belt_List.getExn(1)->int_of_string, tab)
        ),